### PR TITLE
Unusedlevels textbox options

### PR DESCRIPTION
**Pull request by Lunalo**
@africanmathsinitiative/developers 

This is ready for review 

I have added ucrinputtextbox option for checking number of unused levels to be dropped if any and also if there is no unused level then another text will appear to notify the user.

I have also added a color feature in the text box. You are welcome to suggest appropriate color for that .

Thanks.

### CODE REVIEW

**Comment by Lunalo** [comment]
Fixes #3125

File: instat/dlgRemoveUnusedLevels.vb [hunk 1]
```diff
@@ -79,8 +97,31 @@ Public Class dlgRemoveUnusedLevels
         TestOKEnabled()
     End Sub
 
-    Private Sub Controls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverFactorColumn.ControlContentsChanged
+    Private Sub Controls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverFactorColumn.ControlContentsChanged, ucrInputUnusedLevels.ControlContentsChanged
         TestOKEnabled()
     End Sub
 
+    Private Sub ucrReceiverFactorColumn_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrReceiverFactorColumn.ControlValueChanged, ucrRemoveUnusedFactorLevels.ControlValueChanged
+        Dim numOutput As Integer
```
Comment (dannyparsons) [inline instat/dlgRemoveUnusedLevels.vb]: i for integer

File: instat/dlgRemoveUnusedLevels.vb [hunk 2]
```diff
@@ -79,8 +97,31 @@ Public Class dlgRemoveUnusedLevels
...
+            clsSum.AddParameter("x", clsROperatorParameter:=clsTableOperation)
+
+            numOutput = frmMain.clsRLink.RunInternalScriptGetValue(clsSum.ToScript).AsNumeric(0)
+            ucrInputUnusedLevels.txtInput.BackColor = Color.Green
+            If numOutput = 0 Then
+                ucrInputUnusedLevels.SetName("no unused levels to remove")
+            Else
+                ucrInputUnusedLevels.SetName(numOutput & " unused levels will be removed")
+            End If
+        Else
```
Comment (dannyparsons) [inline instat/dlgRemoveUnusedLevels.vb]: clear the text if there is no column

File: instat/dlgRemoveUnusedLevels.vb [hunk 3]
```diff
@@ -79,8 +97,31 @@ Public Class dlgRemoveUnusedLevels
...
+        If Not ucrReceiverFactorColumn.IsEmpty Then
+            clstable.AddParameter("x", clsRFunctionParameter:=clsFactorColumn)
+            clsTableOperation.SetOperation("==")
+            clsTableOperation.AddParameter(clsRFunctionParameter:=clstable, iPosition:=0)
+            clsTableOperation.AddParameter("threshhold", 0, iPosition:=1)
+
+            clsSum.AddParameter("x", clsROperatorParameter:=clsTableOperation)
+
+            numOutput = frmMain.clsRLink.RunInternalScriptGetValue(clsSum.ToScript).AsNumeric(0)
+            ucrInputUnusedLevels.txtInput.BackColor = Color.Green
```
Comment (dannyparsons) [inline instat/dlgRemoveUnusedLevels.vb]: I would prefer the colour to be based on whether there are any levels to remove

**Comment by lilyclements** [comment]
@Lunalo is this ready to re-review?

**Comment by Lunalo** [comment]
@lilyclements  I think you meant @dannyparsons

**Comment by Lunalo** [comment]
@dannyparsons 

This is ready

**Pull request merged**